datafusion/sql/src/statement.rs

```diff
@@ -985,8 +981,52 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
         let inline_constraints = calc_inline_constraints_from_columns(&columns);
         all_constraints.extend(inline_constraints);
 
+        let mut options_map = HashMap::<String, String>::new();
+        for (key, value) in options {
+            if options_map.contains_key(&key) {
+                return plan_err!("Option {key} is specified multiple times");
+            }
+
+            let value_string = match value {
+                Value::SingleQuotedString(s) => s.to_string(),
+                Value::DollarQuotedString(s) => s.to_string(),
+                Value::UnQuotedString(s) => s.to_string(),
+                Value::Number(_, _) | Value::Boolean(_) => value.to_string(),
+                Value::DoubleQuotedString(_)
+                | Value::EscapedStringLiteral(_)
+                | Value::NationalStringLiteral(_)
+                | Value::SingleQuotedByteStringLiteral(_)
+                | Value::DoubleQuotedByteStringLiteral(_)
+                | Value::RawStringLiteral(_)
+                | Value::HexStringLiteral(_)
+                | Value::Null
+                | Value::Placeholder(_) => {
+                    return plan_err!(
+                        "Unsupported Value in CREATE EXTERNAL TABLE statement {}",
+                        value
+                    );
+                }
+            };
+
+            if !(&key.contains('.')) {
+                // If a config does not belong to any namespace, we assume it is
```

Comment:
I am not sure this commet applies to CREATE EXTERNAL TABLE which didn't have format options previously

Can you also please make sure there is test for this alternate syntax (aka `OPTIONS ('has_header' 'true')");` instead of `OPTIONS ('format.has_header' 'true')");` (I may have missed it)